tokio/src/runtime/task/id.rs

```diff
@@ -82,17 +82,23 @@ impl Id {
             crate::loom::lazy_static! {
                 static ref NEXT_ID: StaticAtomicU64 = StaticAtomicU64::new(1);
             }
-            Self(NEXT_ID.fetch_add(1, Relaxed))
+            let id = NonZeroU64::new(NEXT_ID.fetch_add(1, Relaxed))
+                .unwrap_or_else(|| NonZeroU64::new(1).unwrap());
+
+            Self(id)
         }
 
         #[cfg(not(all(test, loom)))]
         {
             static NEXT_ID: StaticAtomicU64 = StaticAtomicU64::new(1);
-            Self(NEXT_ID.fetch_add(1, Relaxed))
+            let id = NonZeroU64::new(NEXT_ID.fetch_add(1, Relaxed))
+                .unwrap_or_else(|| NonZeroU64::new(1).unwrap());
```

Comment:
Please use a loop like this one:

https://github.com/tokio-rs/tokio/blob/0cbf1a5adae81e8ff86ca6d040aeee67cf888262/tokio/src/runtime/task/list.rs#L33-L40

Otherwise it's going to create two threads with id 1 in a row.